salt/_auth/kubernetes_rbac.py

```diff
@@ -1,31 +1,152 @@
 import base64
 import logging
 
+try:
+    import kubernetes.config
+    HAS_KUBERNETES_DEPS = True
+except ImportError:
+    HAS_KUBERNETES_DEPS = False
+
+try:
+    import requests
+    HAS_REQUESTS_DEPS = True
+except ImportError:
+    HAS_REQUESTS_DEPS = False
+
+
 log = logging.getLogger(__name__)
 
 __virtualname__ = 'kubernetes_rbac'
 
 def __virtual__():
-    return __virtualname__
-
-
-ADMIN_USERNAME = b'admin'
-ADMIN_PASSWORD = b'admin'
-ADMIN_TOKEN = base64.encodestring(
-    b'{}:{}'.format(ADMIN_USERNAME, ADMIN_PASSWORD)).rstrip()
-AUTH_HANDLERS = {
-    'basic': {
-        'auth': lambda username, token: all([
-            username == ADMIN_USERNAME,
-            token == ADMIN_TOKEN,
-        ]),
-        'groups': lambda _u, _t: [
-            'node_admin',
-        ],
-    },
+    if not HAS_KUBERNETES_DEPS:
```

Comment:
Small suggestion to avoid masking one of the missing deps if the other one is missing as well:

```python
DEPS_CHECK = {}
try:
    import kubernetes.config
    DEPS_CHECK["python-kubernetes"] = True
except ImportError:
    DEPS_CHECK["python-kubernetes"] = False

try:
    import requests
    DEPS_CHECK["requests"] = True
except ImportError:
    DEPS_CHECK["requests"] = False

...

def __virtual__():
    if not all(DEPS_CHECK.values()):
        return False, 'Missing dependencies: {}'.format(
            ', '.join(dep_name for dep_name, check in DEPS_CHECK.items() if not check)
        )

    return __virtualname__
```

Reply:
I'll apply a variation of this.

Reply:
Somewhat simpler IMHO: https://github.com/scality/metalk8s/pull/978/files#diff-4746d4a429f108ddab90fc4a5c9fcb03R4